components/eventing-controller/testing/safesubscriptions.go

```diff
@@ -0,0 +1,53 @@
+package testing
+
+import (
+	"strings"
+	"sync"
+
+	bebtypes "github.com/kyma-project/kyma/components/eventing-controller/pkg/ems/api/events/types"
+)
+
+// SafeSubscriptions encapsulates Subscriptions to provide mutual exclusion.
+type SafeSubscriptions struct {
+	mutex         *sync.RWMutex
```

Comment:
is there a reason to have the RWMutex as a pointer? With the pointer it is possible to have two different SafeSubscriptions to share the same RWMutex. Having a concrete RWMutex in the struct prevents this. The only thing we need to ensure is, that all the methods are pointer receivers.